### DISCUSSION TITLE
Is it possible to define and register a custom activity but not display it in UI

### DISCUSSION THREAD

**Discussion by defunky** [elsa-workflows/elsa-core#5669] [Q&A]
Can you create and implement a custom activity and not display it in the UI, for example you want it use it as a part of a composite activity but not allow the individual activity to be visible in the designer?

**Comment by glime-ai** [comment]
Yes, you can create and implement a custom activity and choose not to display it in the UI. This can be useful when you want the activity to be used only as part of a composite activity and not be available for individual use in the designer.

To achieve this, you would typically not annotate the custom activity with metadata that the designer uses to list available activities. In Elsa, activities can be annotated with metadata using the `ActivityAttribute` attribute, which includes properties like `Category` and `Description` that the designer uses to display and categorize activities. By omitting this metadata or by setting it in such a way that the designer does not pick it up, you can effectively hide the activity from the UI.

However, it's important to note that this approach relies on the specific implementation details of the workflow engine and its designer UI. The documentation or source code of the workflow engine should be consulted to understand the best way to achieve this behavior.

If you are using Elsa, you might also consider defining the activity in a way that it does not expose any UI hints or descriptors that would make it appear in the designer. Alternatively, you could customize the designer's behavior to filter out certain activities based on your own criteria.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
